weldx/core.py

```diff
@@ -386,10 +388,16 @@ def _initialize_discrete(
 
             # constant value case
             if time is None:
-                time = pd.TimedeltaIndex([0])
+                time = pd.Timedelta(0)
+
+            time = Time(time)
+
+            if time.is_absolute:
+                self._reference_time = time.reference_time
+                time = time - time.reference_time
+
+            time = time.as_pandas_index()
 
-            if isinstance(time, pint.Quantity):
-                time = ut.to_pandas_time_index(time)
             if not isinstance(time, pd.TimedeltaIndex):
                 raise ValueError(
                     '"time" must be a time quantity or a "pandas.TimedeltaIndex".'
```

Comment:
~~~
                )
            dax = xr.DataArray(data=data)
            self._data = dax.rename({"dim_0": "time"}).assign_coords({"time": time})
~~~

(Couldn't mark the lines above, so I added them to avoid confusion)

Maybe it does make sense to create an extra generalized `util.xr_` function for this. This basically just adds a Time object as a dimension to a xarray. Guess we have already used something similar elsewhere?

Reply:
Also, I guess the exception is obsolete if we use the `as_timedeltaindex` from the other PR hera

Reply:
Resolved in PR #445